tests/imputers/test_NearestMeanResponseImputer.py

```diff
@@ -24,56 +33,78 @@ def create_NearestMeanResponseImputer_test_df():
     )
 
 
-class TestInit:
-    """Tests for NearestMeanResponseImputer.__init__."""
+def create_NearestMeanResponseImputer_test_df_2():
+    """Create second DataFrame to use in NearestMeanResponseImputer tests.
 
-    def test_super_init_called(self, mocker):
-        """Test that init calls BaseTransformer.init."""
-        expected_call_args = {
-            0: {"args": (), "kwargs": {"columns": None, "verbose": True}},
-        }
+    DataFrame column c is the response, the other columns are numerical columns containing null entries.
 
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "__init__",
-            expected_call_args,
-        ):
-            NearestMeanResponseImputer(columns=None, verbose=True)
+    """
+    return pd.DataFrame(
+        {
+            "a": [1, 1, np.nan, np.nan, 3, 5],
+            "b": [np.nan, np.nan, 1, 3, 3, 4],
+            "c": [2, 3, 2, 1, 4, 1],
+        },
+    )
 
 
-class TestFit:
-    """Tests for NearestMeanResponseImputer.fit."""
+class TestInit(ColumnStrListInitTests):
+    """Generic tests for transformer.init()."""
 
-    def test_super_fit_called(self, mocker):
-        """Test that fit calls BaseTransformer.fit."""
-        df = create_NearestMeanResponseImputer_test_df()
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "NearestMeanResponseImputer"
 
-        x = NearestMeanResponseImputer(columns=["a", "b"])
 
-        expected_call_args = {
-            0: {
-                "args": (
-                    create_NearestMeanResponseImputer_test_df(),
-                    create_NearestMeanResponseImputer_test_df()["c"],
-                ),
-                "kwargs": {},
-            },
-        }
+class TestFit(GenericFitTests):
+    """Generic tests for transformer.fit()"""
+
+    @classmethod
+    def setup_class(cls):
+        cls.transformer_name = "NearestMeanResponseImputer"
+
+    def test_fit_passed_series(
+        self,
+        initialized_transformers,
+        minimal_dataframe_lookup,
+    ):
+        """Test fit is passed a series as y argument."""
 
-        with ta.functions.assert_function_call(
-            mocker,
-            tubular.base.BaseTransformer,
-            "fit",
-            expected_call_args,
+        df = minimal_dataframe_lookup[self.transformer_name]
+
+        x = initialized_transformers[self.transformer_name]
+
+        with pytest.raises(
+            TypeError,
+            match="unexpected type for y, should be a pd.Series",
         ):
-            x.fit(df, df["c"])
+            x.fit(df)(columns=["c"], separator=333)
+
+    def test_fit_not_changing_data(
```

Comment:
Q: is this one not already contained in a generic class?

Reply:
This one is in the generic class but as per the PR description I moved here the NearestMeanResponseImputer requires a 'y' arg to be passed to it, if you pass it None it'll fall over. The generic class does not pass a 'y' argument  I didn't know whether it was best to change the generic test or to add one in here to overwrite?

Reply:
ahh OK, no this seems fine - again something we might cover in a mixin but in future